spring-cloud-stream-binder-jms-common/src/main/java/org/springframework/cloud/stream/binder/jms/utils/Base64UrlNamingStrategy.java

```diff
@@ -0,0 +1,39 @@
+package org.springframework.cloud.stream.binder.jms.utils;
+
+import java.nio.ByteBuffer;
+import java.util.UUID;
+
+import org.springframework.util.Base64Utils;
+
+/**
+ * Taken from spring-cloud/spring-cloud-stream-binder-rabbit. See
+ * https://github.com/spring-cloud/spring-cloud-stream-binder-rabbit/blob/master/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java#L90
+ *
+ * @author Donovan Muller
+ */
+public class Base64UrlNamingStrategy implements AnonymousNamingStrategy {
+
+    private String prefix = "spring.gen-";
+
+    public Base64UrlNamingStrategy() {
+    }
+
+    public Base64UrlNamingStrategy(String prefix) {
+        this.prefix = prefix;
+    }
+
+    @Override
+    public String generateName() {
+        return generateName(this.prefix);
+    }
+
+    @Override
+    public String generateName(String prefix) {
+        UUID uuid = UUID.randomUUID();
+        ByteBuffer bb = ByteBuffer.wrap(new byte[16]);
+        bb.putLong(uuid.getMostSignificantBits()).putLong(uuid.getLeastSignificantBits());
+        // Convert to base64 and remove trailing =
+        return prefix + Base64Utils.encodeToUrlSafeString(bb.array()).replaceAll("=", "")
+                .replaceAll("-", "");
```

Comment:
No; rabbitmq has no such limitations on queue names.
<img width="231" alt="screen shot 2017-01-09 at 1 01 00 pm" src="https://cloud.githubusercontent.com/assets/483832/21777635/a29b3dda-d66d-11e6-8d51-1ad12b20da9f.png">

Actually, it seems I was conflating two issues; the main reason for this PR is to make header names compliant - I don't think the JMS spec imposes any limitations on queue names - ActiveMQ (at least) allows queues to start with numbers - so it sounds like this was bogus.

Sorry about that.